teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -472,6 +437,59 @@ The official resources will be a small set of resources that are critical in run
 resources will be the Tekton Catalog Maintainers; we will update the contributor ladder for Catalog to reflect that
 contributions are distributed in multiple repositories.
 
+Based on the usefulness of the resources and the bandwidth of the catalog maintainers, the Tekton community has discussed and decided to migrate the following 5 tasks from the current centralized Catalog repository to the `tektoncd-catalog` GitHub organization in the first iteration. Each task is hosted in a separate catalog (repository) and the versioning in the new catalog repository starts with `1.0`:
+
+<table>
+    <thead>
+        <tr>
+            <th>Resource Name</th>
+            <th>Old Version (directory-based)</th>
+            <th>New Catalog Name</th>
+            <th>New Version (git-based)</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan=3><a href="https://github.com/tektoncd/catalog/tree/main/task/git-clone">git-clone</a></td>
+            <td>0.6</td>
+            <td rowspan=3>git-clone</td>
+            <td>1.0</td>
+        </tr>
+        <tr>
+            <td>0.7</td>
+            <td>2.0</td>
+        </tr>
+        <tr>
+            <td>0.8</td>
+            <td>3.0</td>
+        </tr>        
+        <tr>
+            <td rowspan=1><a href="https://github.com/tektoncd/catalog/tree/main/task/golang-build">golang-build</a></td>
+            <td>0.3</td>
+            <td rowspan=2>golang</td>
+            <td>1.0</td>
+        </tr>
+        <tr>
+            <td rowspan=1><a href="https://github.com/tektoncd/catalog/tree/main/task/golang-test">golang-test</a></td>
+            <td>0.2</td>
+            <td>1.0</td>
+        </tr>
+        <tr>
+            <td rowspan=1><a href="https://github.com/tektoncd/catalog/tree/main/task/kaniko">kaniko</a></td>
+            <td>0.6</td>
+            <td>kaniko</td>
+            <td>1.0</td>
+        </tr>
+        <tr>
+            <td rowspan=1><a href="https://github.com/tektoncd/catalog/tree/main/task/git-batch-merge">git-batch-merge</a></td>
+            <td>0.2</td>
+            <td>git-batch-merge</td>
+            <td>1.0</td>
+        </tr>
```

Comment:
just gave a thought, should we move `git-batch-merge` with `git-clone` since this is also used to clone the repos as well as PRs with rebase/merge strategy?
cc @vdemeester @jerop what do you think?
Also a random thought - should we include https://hub.tekton.dev/tekton/task/git-cli as well in this list and keep this in the same catalog as that of git-clone and git-batch-merge?

Reply:
Hi @vinamra28, thanks for the input 😄 . The reason we put `golang-build` and `golang-test` is that they share a lot of common commits (they are "tightly coupled") so it makes sense for them sharing the same release by putting in the same repo, which reduces maintenance cost.

Looking at the commit history of `git-batch-merge` and `git-clone`, they share little common commits. The `git-clone` has a lot more update then `git-batch-merge` so users may see many versions of `git-batch-merge` with no change if we put them in the same repo. I think this is what we want to avoid.

Reply:
I think we should keep `git-clone` and the rest of `git` operation task from each other, mainly because `git-clone` is completely own by us (it uses `git-init`), so it can appear as official in AH, where the other might not.

Reply:
@vdemeester I thought that `git-batch-merge` was also owned by us because
1. our plumbing repo uses that
2. image it's using is our git-init image
😅